### PR TITLE
Gutenboarding: Create public sites when ecommerce

### PR DESCRIPTION
**Pull request by johnHackworth**
This PR changes how the site is created if the user has chosen an e-commerce plan: The site is created as public straight away, so the Atomic transfer after payment work and the user can be redirected to Woo NUX.

### CODE REVIEW

File: client/landing/gutenboarding/components/header/index.tsx [hunk 1]
```diff
@@ -126,10 +128,13 @@ const Header: React.FunctionComponent = () => {
 
 	React.useEffect( () => {
 		if ( newUser && newUser.bearerToken && newUser.username && ! newSite ) {
-			handleCreateSite( newUser.username, newUser.bearerToken );
+			handleCreateSite(
+				newUser.username,
+				newUser.bearerToken,
+				selectedPlan ? selectedPlan.getStoreSlug() : 'free'
```
Comment (p-jackson) [inline client/landing/gutenboarding/components/header/index.tsx:134]:
```suggestion
				selectedPlan ? selectedPlan.getStoreSlug() : 'free_plan'
```

This is the spelling the store uses for the free plan

File: client/landing/gutenboarding/onboarding-block/style-preview/index.tsx [hunk 2]
```diff
@@ -58,9 +59,10 @@ const StylePreview: React.FunctionComponent = () => {
 
 	const handleCreateSite = useCallback(
 		( username: string, bearerToken?: string ) => {
-			createSite( username, freeDomainSuggestion, bearerToken );
+			const planSlug = selectedPlan ? selectedPlan.getStoreSlug() : 'free';
```
Comment (p-jackson) [inline client/landing/gutenboarding/onboarding-block/style-preview/index.tsx:62]:
```suggestion
			const planSlug = selectedPlan ? selectedPlan.getStoreSlug() : 'free_plan';
```

File: client/landing/gutenboarding/stores/onboard/actions.ts [hunk 3]
```diff
@@ -99,10 +101,12 @@ export function* createSite(
 
 	const currentDomain = domain ?? freeDomainSuggestion;
 	const siteUrl = currentDomain?.domain_name || siteTitle || username;
+	const isPublicSite = planSlug === 'ecommerce-bundle';
```
Comment (p-jackson) [inline client/landing/gutenboarding/stores/onboard/actions.ts:104]: The store is importing plan slugs from here.

```typescript
import {
	PLAN_FREE,
	PLAN_PERSONAL,
	PLAN_PREMIUM,
	PLAN_BUSINESS,
	PLAN_ECOMMERCE,
} from '../../../../lib/plans/constants';
```

We should probably be doing the same here, and in other places where we hard code plan slugs.

**Comment by p-jackson** [comment]
Ah I see that the plans grid is available behind a feature flag.

**Comment by ramonjd** [comment]
Selecting an e-commerce flow from the plans grid overlay in Gutenboarding creates a public site ✅ 

Successfully purchased an e-commerce plan at checkout, went through the store setup flow, and an Atomic site was created ✅ 

> Looks like this will probably work once the plan grid is part of the gutenboarding flow. 

👍

**Comment by p-jackson** [comment]
Tested. This is good to go 🚢

**Pull request merged**